server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManagerImpl.java

```diff
@@ -306,6 +312,111 @@ public boolean mkdirs(Path path, FsPermission permission) throws IOException {
     return getVolumeByPath(path).getFileSystem().mkdirs(path, permission);
   }
 
+  @Override
+  public boolean mkdirs(Path path, String storagePolicy, String encoding) throws IOException {
+    boolean ret = mkdirs(path);
+    try {
+      checkDirPolicies(path, storagePolicy, encoding);
+    } catch (IOException e) {
+      // non-fatal error, just print warning and continue
+      // directory will just have wrong policy.
+      log.warn("error checking policy for " + path, e);
+    }
+    return ret;
+  }
+
+  @Override
+  public void checkDirPolicies(Path path, String storagePolicy, String encoding)
+      throws IOException {
+    FileSystem fs = getVolumeByPath(path).getFileSystem();
+    // check that path exists...it may not yet, which is ok. just return
+    if (!fs.exists(path)) {
+      log.debug("check {}: path does not exist", path);
+      return;
+    }
+    if (fs instanceof DistributedFileSystem) {
+      DistributedFileSystem dfs = (DistributedFileSystem) fs;
+      BlockStoragePolicySpi currPolicy = dfs.getStoragePolicy(path);
```

Comment:
Do you know anything about the expense of these methods?  Will these cause RPCs to namenode?

Reply:
will look into it, but I'm hoping this isn't called all that often.  should be just at tablet creation time, or when doing a bulk import.

Reply:
@keith-turner yes, it looks like a call to the namenode is made to get the policy info.  I removed some code from Tablet that might have caused some problems, so now this should only be called when a directory is created, or when a user runs the conversion tool I added.